diff --git a/src/Plugin/Field/FieldFormatter/AceFormatter.php b/src/Plugin/Field/FieldFormatter/AceFormatter.php
index a71fef05b1459f05b0f855117815d928c6c0e263..ac33a71415fe63ecd5af85e8fe9186b8f6e2de91 100644
--- a/src/Plugin/Field/FieldFormatter/AceFormatter.php
+++ b/src/Plugin/Field/FieldFormatter/AceFormatter.php
@@ -91,7 +91,7 @@ class AceFormatter extends FormatterBase implements ContainerFactoryPluginInterf
    */
   public static function defaultSettings() {
     // Get default ace_editor configuration.
-    $config = $this->configFactory->get('ace_editor.settings');
+    $config = \Drupal::config('ace_editor.settings')->get();
     return $config + parent::defaultSettings();
   }
 
